### ISSUE TITLE
Fix invalid XML with multiple item attributes from XPath query results

### ISSUE THREAD

**Issue by mlcollard** [srcML/srcML#1949]
The attribute `item` is added to query results, such as from XPath, to make each result unique. For example, the unit for the file nm.cpp contains the following unit:

```XML
<unit xmlns="http://www.srcML.org/srcML/src" revision="1.0.0" language="C++" filename="nm.cpp">
<expr_stmt><expr><name>n</name> <operator>=</operator> <name>m</name></expr>;</expr_stmt>
</unit>
```

If we run an XPath query on this unit for names, i.e., `srcml nm.cpp --xpath="//src:name"`, we get two results with the same filename. The attribute `item` is to allow each to be unique:

```XML
<unit xmlns="http://www.srcML.org/srcML/src" revision="1.0.0">

<unit revision="1.0.0" language="C++" filename="nm.cpp" item="1"><name>n</name></unit>

<unit revision="1.0.0" language="C++" filename="nm.cpp" item="2"><name>m</name></unit>

</unit>
```

The problem is when a query is run on this, then we get two item attributes, which is invalid XML:

```XML
<unit xmlns="http://www.srcML.org/srcML/src" revision="1.0.0">

<unit revision="1.0.0" language="C++" filename="nm.cpp" item="1" item="1"><name>n</name></unit>

<unit revision="1.0.0" language="C++" filename="nm.cpp" item="2" item="1"><name>m</name></unit>

</unit>
```

In this case, the second item attribute is unnecessary because each unit from the first query results in a single unit from the second query. So the second item attribute should not occur.

So what if a unit from the first query results in multiple units in a second query? In that case, the behavior should be to append the second query item result onto the first. In this case, by using a period.

First query result of `srcml nm.cpp --xpath="//src:expr"`:

```XML
<unit xmlns="http://www.srcML.org/srcML/src" revision="1.0.0">

<unit revision="1.0.0" language="C++" filename="nm.cpp" item="1"><expr><name>n</name> <operator>=</operator> <name>m</name></expr></unit>

</unit>
```

Second query result of `srcml --xpath="//src:name"`, where repeated applications of the same query would not change the results:

```XML
<unit xmlns="http://www.srcML.org/srcML/src" revision="1.0.0">

<unit revision="1.0.0" language="C++" filename="nm.cpp" item="1.1"><name>n</name></unit>

<unit revision="1.0.0" language="C++" filename="nm.cpp" item="1.2"><name>m</name></unit>

</unit>
```

To summarize: The current behavior is to add item attributes for every query result, leading to invalid XML. The new behavior:

* Add an item attribute if one is not present
* If an item attribute is present, and the current unit produced only one output result unit, leave the unit attribute alone
* If an item attribute is present, and the current unit produced multiple output result units, append a period '.' and another level of counting to the current item attribute.

There is an alternative to renumber the item attribute each time. However, this loses which unit in the source produced the unit in the result.
